torch_xla/utils/utils.py

```diff
@@ -25,7 +25,14 @@ def __init__(self):
 
 
 class SampleGenerator(object):
-  """SampleGenerator (PLACEHOLDER)"""
+  """Iterator which returns multiple samples of a given input data.
+
+  Can be used in place of a PyTorch `DataLoader` to generate synthetic data.
+
+  Args:
+    data: The input data which should be returned.
```

Comment:
This one is a bit vague- I think data should be a sample tensor and this method will return tensors of the same shape and dtype filled in with random values.

Could be good to give an example like:
(torch.zeros(FLAGS.batch_size, 3, img_dim, img_dim),
 torch.zeros(FLAGS.batch_size, dtype=torch.int64))

Or we could point to a sample usage:
https://github.com/pytorch/xla/blob/master/test/test_train_imagenet.py#L107

Reply:
They will not be returned with random values ... but the very same exact data will be returned num_samples times.

Reply:
Links to code tend to be broken if code changes.
I tried to add a literal block (https://devguide.python.org/documenting/#source-code) but google-pyformat makes a mess out of it.